addons/account/static/src/js/account_selection.js

```diff
@@ -1,100 +1,31 @@
-                    kwargs: {
-                        domain: self.record.context && self.record.context.account_type_domain || [],
-                        fields: ['id', 'internal_group', 'display_name'],
-                    },
-                }).then(function(arg) {
-                    self.values = _.map(arg, v => [v['id'], v['display_name']]);
-                    var unique_groups = _.uniq(_.map(arg, v => v['internal_group']));
-                    if (unique_groups.length > 1) {
-                        var parent_lookup = {
-                            'bs': {'name': _t('Balance Sheet'), 'index': 0},
-                            'pl': {'name': _t('Profit & Loss'), 'index': 1},
-                        }
-                        var lookup = {
-                            'asset': {'parent': 'bs', 'name': _t('Assets')},
-                            'liability': {'parent': 'bs', 'name': _t('Liabilities')},
-                            'equity': {'parent': 'bs', 'name': _t('Equity')},
-                            'income': {'parent': 'pl', 'name': _t('Income')},
-                            'expense': {'parent': 'pl', 'name': _t('Expense')},
-                        }
-                        var unique_parents = _.uniq(_.map(_.filter(unique_groups, u => u in lookup), g => lookup[g].parent));
-                        self.hierarchy_groups = [];
-                        _.each(unique_parents, parent_id => {
-                            var parent = parent_lookup[parent_id];
-                            self.hierarchy_groups.splice(parent.index, 0, {'name': parent.name, 'children': []});
-                        });
-                        var other_accounts_added = false;
-                        _.each(unique_groups, g => {
-                            if (g in lookup){
-                                var rec = lookup[g];
-                                self.hierarchy_groups[parent_lookup[rec.parent].index].children.push({
-                                    'name': rec['name'],
-                                    'ids': _.map(_.filter(arg, v => v['internal_group'] == g), v => v['id'])
-                                });
-                            } else if (!other_accounts_added){
-                                self.hierarchy_groups.push({
-                                    'name': _t('Other'),
-                                    'ids': _.map(_.filter(arg, v => !_.keys(lookup).includes(v['internal_group'])), v => v['id'])
-                                });
-                                other_accounts_added = true;
-                            }
-                        })
-                    }
-                });
-            }
+    init: function () {
+        this._super.apply(this, arguments);
+        this.hierarchyGroups = [
+            {name: _t('Balance Sheet')},
+            {name: _t('Assets'), children: this.values.filter(x => x[0] && x[0].startsWith('asset_'))},
```

Comment:
My bad, I thought that all the types where now starting with type_ but there are three values that don't follow that format (namely "equity, income and expense") so we don't propose those 3 values by splitting on `equity_, income_ and expense_`. You can remove the _ from startsWith('type_') that way it will display all possible values :)